nomad/event_sink_manager.go

```diff
@@ -217,15 +222,15 @@ func (m *SinkManager) updateSinkProgress() error {
 	return nil
 }
 
+// removeSink removes a sink from the manager's subscriptions.
+// removeSink should only be called from Run
 func (m *SinkManager) removeSink(id string) {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-
 	delete(m.sinkSubscriptions, id)
 }
 
 // refreshSinks checks for any new event sinks added to the state store. It
-// adds new sinks as new ManagedSinks
+// adds new sinks as new ManagedSinks. This method must be externally
+// synchronized in the SinkMangers main run loop
```

Comment:
Minor docstring nitpick, just to make sure it's totally unambiguous with with the other `run` method in this file:

```suggestion
// synchronized in the SinkManger.Run main loop
```